src/Workspaces/CoreTest/SolutionTests/ProjectDependencyGraphTests.cs

```diff
@@ -68,45 +98,254 @@ private void VerifyDependencySets(string projectReferences, string expectedResul
 
-        private void VerifyReverseTransitiveReferences(string projectReferences, string project, string[] expectedResults)
+        [Fact, Trait(Traits.Feature, Traits.Features.Workspace)]
+        public void TestGetProjectsThatTransitivelyDependOnThisProjectThrowsArgumentNull()
+        {
+            var solution = CreateSolutionFromReferenceMap("");
+
+            Assert.Throws<ArgumentNullException>("projectId",
+                () => solution.GetProjectDependencyGraph().GetProjectsThatTransitivelyDependOnThisProject(null));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Workspace)]
+        public void TestReverseTransitiveReferencesIncrementalUpdateInMiddle()
+        {
+            // We are going to create a solution with the references:
+            //
+            // A -> B -> C -> D
+            //
+            // but we will add the B -> C link last, to verify that when we add the B to C link we update the reverse references of D.
+
+            var solution = CreateSolutionFromReferenceMap("A B C D");
+            VerifyReverseTransitiveReferences(solution, "A", new string[] { });
+            VerifyReverseTransitiveReferences(solution, "B", new string[] { });
+            VerifyReverseTransitiveReferences(solution, "C", new string[] { });
+            VerifyReverseTransitiveReferences(solution, "D", new string[] { });
+
+            solution = AddProjectReferences(solution, "A", new string[] { "B" });
+            solution = AddProjectReferences(solution, "C", new string[] { "D" });
+
+            VerifyDirectReverseReferences(solution, "B", new string[] { "A" });
+            VerifyDirectReverseReferences(solution, "D", new string[] { "C" });
+
+            VerifyReverseTransitiveReferences(solution, "A", new string[] { });
+            VerifyReverseTransitiveReferences(solution, "B", new string[] { "A" });
+            VerifyReverseTransitiveReferences(solution, "C", new string[] { });
+            VerifyReverseTransitiveReferences(solution, "D", new string[] { "C"});
+
+            solution = AddProjectReferences(solution, "B", new string[] { "C" });
+
+            VerifyDirectReverseReferences(solution, "C", new string[] { "B" });
+
+            VerifyReverseTransitiveReferences(solution, "A", new string[] { });
+            VerifyReverseTransitiveReferences(solution, "B", new string[] { "A" });
+            VerifyReverseTransitiveReferences(solution, "C", new string[] { "A", "B" });
+            VerifyReverseTransitiveReferences(solution, "D", new string[] { "A", "B", "C" });
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Workspace)]
+        public void TestReverseTransitiveReferencesForUnrelatedProjectAfterWithProjectReferences()
+        {
+            // We are going to create a solution with the references:
+            //
+            // A -> B       C -> D
+            //
+            // and will then remove the reference from B to C. This process will cause us to throw out
```

Comment:
Should this say "will then remove the reference from **C to D**" and on the next line "asking for the reverse references of **B**"?